tool-testsuite/test/org/antlr/v4/test/tool/TestVocabulary.java

```diff
@@ -64,7 +67,7 @@ public void testVocabularyFromTokenNames() {
 		Vocabulary vocabulary = VocabularyImpl.fromTokenNames(tokenNames);
 		Assert.assertNotNull(vocabulary);
 		Assert.assertEquals("EOF", vocabulary.getSymbolicName(Token.EOF));
-		for (int i = 0; i < tokenNames.length; i++) {
+		for (int i = Token.MIN_USER_TOKEN_TYPE; i <= vocabulary.getMaxTokenType(); i++) {
```

Comment:
❗️ The original test was correct. We want to test the full range of the original array. If you want, you could add an assertion that `vocabulary.getMaxTokenType()` matches `tokenNames.length - 1`.